aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -1227,4 +1196,16 @@ private AuthCodeDeliveryDetails convertCodeDeliveryDetails(UserCodeDeliveryDetai
                     details.getAttributeName())
             : null;
     }
+
+    private boolean isUserAttributeVerifiedByCode(String userAttributeKey) {
+        if (userAttributeKey.equals(AuthUserAttributeKey.email().getKeyString())) {
+            return true;
+        }
+
+        if (userAttributeKey.equals(AuthUserAttributeKey.phoneNumber().getKeyString())) {
```

Comment:
It is better practice to write:
```java
if (definitelyNotNull.equals(possiblyNull)) {
    // No NullPointerException!
}
```
Than:
```java
if (possiblyNull.equals(definitelyNotNul)) {
    // Uh oh, might have NPE
}
```

Here, I think `AuthUserAttributeKey.phoneNumber().getKeyString()` is always non null -- but what about `userAttributeKey`? 

(Same comment for `if` check above)

------

Other thought -- you could do like:
```java
List<String> codeVerifiedAttributeKeys = Arrays.asList(
    AuthUserAttributeKey.phoneNumber().getKeyString(),
    AuthUserAttributeKey.email().getKeyString()
);
return codeVerifiedAttributeKeys.contains(/* provided */ userAttributeKey);
```

4 lines instead of 10 🤷‍♂️

Reply:
I got your idea. Yeah I should always avoid any possible NPE. 😭